src/main/kotlin/no/nav/familie/ba/sak/kjerne/vedtak/domene/VedtaksperiodeMedBegrunnelser.kt

```diff
@@ -179,19 +180,24 @@ fun VedtaksperiodeMedBegrunnelser.hentUtbetalingsperiodeDetaljer(
         Vedtaksperiodetype.ENDRET_UTBETALING,
         -> {
             val utbetalingsperioderRelevantForVedtaksperiode =
-                utbetalingsperiodeDetaljer.perioder().find { andelerVertikal ->
-                    andelerVertikal.fraOgMed.tilFørsteDagIMåneden().tilLocalDate()
-                        .isSameOrBefore(this.fom ?: TIDENES_MORGEN) &&
-                        andelerVertikal.tilOgMed.tilSisteDagIMåneden().tilLocalDate()
-                            .isSameOrAfter(this.tom ?: TIDENES_ENDE)
-                }?.innhold ?: throw Feil(
+                finnUtbetalingsperioderRelevantForVedtaksperiode(utbetalingsperiodeDetaljer) ?: throw Feil(
                     "Finner ikke segment for vedtaksperiode (${this.fom}, ${this.tom}) blant segmenter ${andelerTilkjentYtelse.utledSegmenter()}",
                 )
 
             utbetalingsperioderRelevantForVedtaksperiode.toList()
```

Comment:
Nå trenger vi ikke å dra det ut i en egen variabel tenker jeg

```suggestion
            finnUtbetalingsperioderRelevantForVedtaksperiode(utbetalingsperiodeDetaljer)?..toList() ?: throw Feil(
                    "Finner ikke segment for vedtaksperiode (${this.fom}, ${this.tom}) blant segmenter ${andelerTilkjentYtelse.utledSegmenter()}",
                )
```

Reply:
Godt poeng!